### DISCUSSION TITLE
GPIO Eingang zum Ein/Aus Schalten des Displays

### DISCUSSION THREAD

**Discussion by Tueftler1983** [tbnobody/OpenDTU#1885] [Ideas]
Hallo ist es möglich einen GPIO Eingang zum Anschluss eines Tasters/Bewegungsmelders zu integrieren um das angeschlossene Display Ein/Aus zu schalten??

Wäre schön wenn dies umgesetzt werden könnte

**Comment by Marc9485** [comment]
Taster oder Schalter an die - Leitung keine Option?

**Reply by Tueftler1983** [reply]
Denke das funktioniert nicht, kenne es von easyESP nur so das das Display bein booten komplett angeschlossen sein muss damit es initialisiert werden kann

**Reply by Tiese** [reply]
Hallo @Tueftler1983,
ich habe einfach eine 2. pinmapping mit den Werten für das Display auf -1 gesetzt.
Wenn ich es aus, bzw. einschalten möchte, lade ich die entsprechende pinmapping-Datei - fertig :-)

Schöne Grüße
Tiese

**Reply by ghost** [reply]
Ich mische mich noch mal ein.

Wenn es die Zeit erlaubt, kommt von mir evtl. noch dieses Jahr eine abgewandelte openDTU Variante.
Diese wird z.b. eine Unterstützung für[ umgebaute I2C LCDs](https://github.com/tbnobody/OpenDTU/discussions/1640) zur Steuerung der Hintergrundbeleuchtung besitzen.
An dem originalem LCD geht beim nächtlichen Stand-by normalerweise das Hintergrundlicht nicht aus.
Das macht dann aber auch eine angepasste Pin_Mapping-Datei erforderlich. Bei mir läuft das schon.
Eigentlich wünsche ich noch andere Anpassungen, dazu mehr, wenn es soweit ist.

Ich könnte eine BWM-/ Dämmerungs-/ Schalteransteuerung hinzufügen.
Auch die nötigen Pin-Einstellungen oder zusätzliche Steuerelemente in der Web-App bekomme ich hin.
Was benötigt ihr denn genau?
Soll das Display nur auf den Eingang reagieren?
Soll der nächtliche Stromsparmodus berücksichtigt werden?
Tastfunktion, Schalter, beides?
...
Ihr müsst dann nur noch etwas Geduld haben. Freizeit ist bei mir knapp. Ich frage mich manchmal, wie die Entwickler hier das hinbekommen.

**Comment by ghost** [comment]
Das ginge schon, die Zauberfunktion ist: "display.begin". Das müsste der openDTU nur mitgeteilt werden, allerdings würde sich damit warscheinlich das Spannungabschalten am Display erübrigen, bzw. wäre parallel dazu.
Das Display muss nicht nur beim openDTU-Start sondern auch bei dieser Funktion initialisiert werden, das ist an sich kein Problem, das erfolgt jetzt schon beim Deaktivieren des Stromsparmoduses.
Ich denke nur nicht, das es so eine Funktion in die offizielle Firmware schafft. Man müsste es halt selbst integrieren.

Man bräuchte mindestens einen Button in den Einstellungen, der das Display dauerhaft ausknippst; sprich nicht nur, wenn kein Wechselrichter erreichbar ist.
Es wäre ein zusätlicher Pin als Eingang zu definieren, der das Display initialisiert.
- Und Variante 1: zeitbegrenzt per Impuls (Taster oder BWM) einschaltet.
- Oder Variante 2: nur initialisiert, ausgeknipst wird es per BWM.
- Oder Variante 3. Toogle-Betrieb
- ...
Das würde ich für mich schon hinbekommen.

Ich habe mir für meine eigene openDTU auch einen zusätzlichen PIin und einen Schieberegler zur Hintergrundbeleuchtung des LCD-Display hinzugefügt. (Muss vom Display unterstützt werden. Ich lade auf meinen Account irgendwann mal noch eine Umbauanleitung für ein I2C-LCD hoch, die ist [hier](https://github.com/tbnobody/OpenDTU/discussions/1640) schon zu finden. )
Der extra Pin war/ist nötig, da die die Standardoption nur die Schrift ausblendet. 
Der Schieberegler ist für die Helligkeit der Hintergrundbeleuchtung.
Das liegt daran, das ursprünglich nur OLEDs unterstützt wurden und die halt keine Hintergrundbeleuchtung benötigen. Weiterhin ist jeder  verwendete Pin anderweitig nicht mehr verfügbar und je nach verrwendeten ESP ist die aAnzahl der Pins begrenzt.
Genau so geht mir das "grüne Lebenzeichen" auf den Senkel. Bei mir ist da nur ein nicht so auffäliger kleiner grüner Punkt. Bei Nichtereichbarkeit der openDTU Hardware ist das rote Sympol animiert. Nichterreichbarkeit ist ja viel interessanter. Ich habe die Animation sozusagen invertiert.

Allerdings kämpfe ich, sobalt eine neue Version von openDTU heraus kommt, jedesmal damit, das die neue Version in meinen Fork integriert wird, das klappt nicht immer ganz reibungslos. Ich komme auch nur mit viel Mühe und auch sehr unzureichend mit den ganzen Git-Funktionen klar.
Da ich diese Funktion nicht benötige und wie gesagt zu wenig Ahnung habe, wird es daher von mir keine angepasste openDTU-Version geben.
